op-node/rollup/derive/engine_controller.go

```diff
@@ -412,8 +451,8 @@ func (e *EngineController) InsertUnsafePayload(ctx context.Context, envelope *et
 	}
 
 	//update unsafe,safe,finalize and send fcu for sync
-	if status.Status == eth.ExecutionInconsistent {
-		log.Info("engine meet inconsistent here")
+	if needSyncWithEngine {
+		log.Info("engine meet inconsistent, sync status")
```

Comment:
It is not appropriate to assume that both safe and finalized are already equal to unsafe here. It is recommended to add a condition to check if safe and finalized are greater than unsafe before proceeding with the override; otherwise, it is unnecessary.

Reply:
updated